src/test/java/com/mael/payroll/paymentMethods/DepositMethodTest.java

```diff
@@ -0,0 +1,15 @@
+package com.mael.payroll.paymentMethods;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class DepositMethodTest {
+
+    @Test
+    public void createsADepositMethod() {
```

Comment:
Would something like `hasBankAndAccount` fit better here?
In the end you're not verifying the creation of a deposit method here, are you?